python/ray/tune/tune.py

```diff
@@ -238,15 +245,17 @@ def run(run_or_experiment,
 
     runner.add_experiment(experiment)
 
-    if verbose:
-        print(runner.debug_string(max_debug=99999))
+    if IS_NOTEBOOK:
+        reporter = JupyterNotebookReporter(verbosity=verbose)
+    else:
+        reporter = CLIReporter()
```

Comment:
I think we need to feature flag this; the logic here should be:

```
if IS_NOTEBOOK:
    reporter = JupyterNotebookReporter(verbosity=verbose)
elif verbose == 2 or tabulate not installed:  # assuming 2 is default
    logger.info("To try out the tabular output by setting `tune.run(verbose=1...)`")
    reporter = THE_STATUS_QUO_Reporter()
else:
    reporter = CLIReporter()

Reply:
Actually, maybe instead of `THE_STATUS_QUO_Reporter`, let's look at this, from trial.py:progress_string:

```

    def progress_string(self):
        """Returns a progress message for printing out to the console."""

        if not self.last_result:
            return self._status_string()

        def location_string(hostname, pid):
            if hostname == os.uname()[1]:
                return "pid={}".format(pid)
            else:
                return "{} pid={}".format(hostname, pid)

        pieces = [
            "{}".format(self._status_string()), "[{}]".format(
                self.resources.summary_string()), "[{}]".format(
                    location_string(
                        self.last_result.get(HOSTNAME),
                        self.last_result.get(PID))), "{} s".format(
                            int(self.last_result.get(TIME_TOTAL_S, 0)))
        ]

        if self.last_result.get(TRAINING_ITERATION) is not None:
            pieces.append("{} iter".format(
                self.last_result[TRAINING_ITERATION]))

        if self.last_result.get(TIMESTEPS_TOTAL) is not None:
            pieces.append("{} ts".format(self.last_result[TIMESTEPS_TOTAL]))

        if self.last_result.get(EPISODE_REWARD_MEAN) is not None:
            pieces.append("{} rew".format(
                format(self.last_result[EPISODE_REWARD_MEAN], ".3g")))

        if self.last_result.get(MEAN_LOSS) is not None:
            pieces.append("{} loss".format(
                format(self.last_result[MEAN_LOSS], ".3g")))

        if self.last_result.get(MEAN_ACCURACY) is not None:
            pieces.append("{} acc".format(
                format(self.last_result[MEAN_ACCURACY], ".3g")))

        return ", ".join(pieces)

```

Can you make sure all the same information (i.e., the same fields) show up on the table? This will make sure that any information that we could previously get via the old output format is not lost.

Reply:
Got rid of the `progress_string` func and moved the logic out to `progress_reporter`